esmvalcore/_recipe.py

```diff
@@ -495,15 +495,17 @@ def _update_fx_settings(settings, variable, config_user):
     for step in ('area_statistics', 'volume_statistics', 'zonal_statistics'):
         var = dict(variable)
         fx_files_dict = {}
-        if settings.get(step, {}).get('fx_files'):
-            var['fx_files'] = settings.get(step, {}).get('fx_files')
+        var['fx_files'] = settings.get(step, {}).get('fx_files')
+        if var['fx_files']:
             for fxvar in var['fx_files']:
                 fxvar_name = fxvar
                 if isinstance(fxvar, dict):
-                    fxvar_name = fxvar.get("short_name")
-                    if not fxvar_name:
+                    try:
+                        fxvar_name = fxvar["short_name"]
+                    except KeyError:
                         raise RecipeError(
-                            "Key short_name missing from {}".format(fxvar))
+                            "No valid short_name found in {}".format(fxvar)
+                        )
```

Comment:
Sorry for not having been clear enough. I presented two alternatives, now we have a mixture that is again a bit unclear since actually with the try-except an invalid but present entry could slip by.

So to clarify: Either catch `KeyError` -> "No entry found" OR test for validity -> "No valid name found".

Reply:
bit confused, how does it look now?